app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchSuggestionsFragment.kt

```diff
@@ -126,14 +176,36 @@ class SearchSuggestionsFragment : Fragment(), CoroutineScope {
 
 @Composable
 private fun SearchOverlay(
-    viewModel: SearchSuggestionsViewModel
+    viewModel: SearchSuggestionsViewModel,
+    topSites: List<TopSite>,
+    topSitesMenuItems: List<TopSiteMenuItem>,
+    onTopSiteClick: (TopSite) -> Unit = {}
 ) {
     val state = viewModel.state.observeAsState()
+    val query = viewModel.searchQuery.observeAsState()
```

Comment:
Now that this whole tree will get recomposed for every text change, maybe we should pass down the text to `SearchSuggestions()` to not have to subscribe there again? 🤔